File: Library-Client-App - aul/src/main/java/co/id/library/project/LibraryClientApp/models/History.java

package co.id.library.project.LibraryClientApp.models;

import java.util.Date;
import lombok.Data;

@Data
public class History {
    private Integer idHistory;
    private Integer idProject;
    private String info;
    private Date waktu;
    private String pesan;
    private String status;

    public History() {
    }

    public History(Integer idHistory, Integer idProject, String info, Date waktu, String pesan, String status) {
        this.idHistory = idHistory;
        this.idProject = idProject;
        this.info = info;
        this.waktu = waktu;
        this.pesan = pesan;
        this.status = status;
    }
    
}

File: Library-Client-App - zaki/src/main/resources/static/js/trainee/history.js
let myHistory = new Object();
let table = null;

$(document).ready(() => {
    getAll();
});

function getAll() {
    table = $('#myHistory').DataTable({
        filter: true,
        orderMulti: true,
        ajax: {
            url: "/history/trainee",
            datatype: "json",
            dataSrc: ""
        },
        columns: [
            {
                data: "id", name: "No", autoWidth: true,
                render: function (data, type, row, meta) {
                    return meta.row + meta.settings._iDisplayStart + 1;
                }
            },
            {
                data: "idProject", name: "Project", autoWidth: true
            },
            {
                data: "info", name: "Batch", autoWidth: true
            },
            {
                data: "waktu", name: "Nama Trainee", autoWidth: true,
                render: function (data) {
                    return moment(data).format('ddd, MMMM Do, YYYY');
                }
            },
            {
                data: "pesan", name: "Nama Trainer", autoWidth: true
            },
            {
                data: "status", name: "Deskripsi", autoWidth: true
            }
            
        ]
    });
}